wandb/sdk/lib/sock_client.py

```diff
@@ -149,20 +153,48 @@ def send_record_publish(self, record: "pb.Record") -> None:
         server_req.record_publish.CopyFrom(record)
         self.send_server_request(server_req)
 
+    def _buffer_get(self, start: int, end: int, advance: bool = False) -> bytes:
+        if advance:
+            data = b"".join(self._buffer_list)
+            requested = data[start:end]
+            leftover = data[end:]
```

Comment:
rm intermediate step to reduce memory consumption?

Reply:
yeah, the if advance is not fully otpimized.. the fall throguh case is, trying to figure out how to unify them